internal/cmp/resource_network.go

```diff
@@ -138,12 +142,24 @@ func (r *resNetwork) Delete(ctx context.Context, d *utils.Data, meta interface{}
 	return err
 }
 
-func alignNetworkReq(request *models.CreateNetwork) {
-	request.Site.ID = request.GroupID
-	if request.NetworkDomainID != 0 {
-		request.NetworkDomain = &models.IDModel{ID: request.NetworkDomainID}
+func (r *resNetwork) networkRequest(createReq *models.CreateNetwork) error {
+	if createReq.TfDhcpNetwork != nil {
+		createReq.DhcpServer = createReq.TfDhcpNetwork.DhcpServer
+		if createReq.TfDhcpNetwork.Config != nil {
+			createReq.Config.ConnectedGateway = createReq.TfDhcpNetwork.Config.ConnectedGateway
+			createReq.Config.VlanIDs = createReq.TfDhcpNetwork.Config.VlanIDs
+			createReq.Config.SubnetDhcpLeaseTime = createReq.TfDhcpNetwork.Config.SubnetDhcpLeaseTime
+			createReq.Config.SubnetDhcpServerAddress = createReq.TfDhcpNetwork.Config.SubnetDhcpServerAddress
+			createReq.Config.SubnetIPManagementType = createReq.TfDhcpNetwork.Config.SubnetIPManagementType
+			createReq.Config.SubnetIPServerID = createReq.TfDhcpNetwork.Config.SubnetIPServerID
+		}
 	}
-	if request.ProxyID != 0 {
-		request.NetworkProxy = &models.IDModel{ID: request.ProxyID}
+	if createReq.TfStaticNetwork != nil {
```

Comment:
why did you remove if request.ProxyID != 0 {
		request.NetworkProxy = &models.IDModel{ID: request.ProxyID}

Reply:
updated